--- FreeRDP-8d0809cf26868fc5ff651850a32724650597747f/winpr/libwinpr/crypto/cipher.c~	2016-12-19 12:14:30.000000000 +0000
+++ FreeRDP-8d0809cf26868fc5ff651850a32724650597747f/winpr/libwinpr/crypto/cipher.c	2016-12-27 18:46:04.465261959 +0000
@@ -26,7 +26,6 @@
 
 #ifdef WITH_OPENSSL
 #include <openssl/aes.h>
-#include <openssl/rc4.h>
 #include <openssl/des.h>
 #include <openssl/evp.h>
 #endif
@@ -50,12 +49,7 @@
 	if (!key || (keylen == 0))
 		return NULL;
 
-#if defined(WITH_OPENSSL)
-	if (!(ctx = (WINPR_RC4_CTX*) calloc(1, sizeof(RC4_KEY))))
-		return NULL;
-	RC4_set_key((RC4_KEY*) ctx, keylen, key);
-
-#elif defined(WITH_MBEDTLS) && defined(MBEDTLS_ARC4_C)
+#if defined(WITH_MBEDTLS) && defined(MBEDTLS_ARC4_C)
 	if (!(ctx = (WINPR_RC4_CTX*) calloc(1, sizeof(mbedtls_arc4_context))))
 		return NULL;
 	mbedtls_arc4_init((mbedtls_arc4_context*) ctx);
@@ -66,11 +60,7 @@
 
 BOOL winpr_RC4_Update(WINPR_RC4_CTX* ctx, size_t length, const BYTE* input, BYTE* output)
 {
-#if defined(WITH_OPENSSL)
-	RC4((RC4_KEY*) ctx, length, input, output);
-	return TRUE;
-
-#elif defined(WITH_MBEDTLS) && defined(MBEDTLS_ARC4_C)
+#if defined(WITH_MBEDTLS) && defined(MBEDTLS_ARC4_C)
 	if (mbedtls_arc4_crypt((mbedtls_arc4_context*) ctx, length, input, output) == 0)
 		return TRUE;
 #endif
@@ -82,11 +72,7 @@
 	if (!ctx)
 		return;
 
-#if defined(WITH_OPENSSL)
-	memset(ctx, 0, sizeof(RC4_KEY));
-	free(ctx);
-
-#elif defined(WITH_MBEDTLS) && defined(MBEDTLS_ARC4_C)
+#if defined(WITH_MBEDTLS) && defined(MBEDTLS_ARC4_C)
 	mbedtls_arc4_free((mbedtls_arc4_context*) ctx);
 	free(ctx);
 #endif
